src/submit.py

```diff
@@ -121,6 +120,57 @@ def create_array_input_file(input_dir, output_dir, job_name, tmp_dir):
     return tmp_file_array
 
 
+def estimate_job_time(input_path: str) -> str:
+    """
+    Estimate total run time based on input file/folder
+
+    Parameters
+    ----------
+    input_path: str
+        Input audio file or folder containing audio files.
+
+    Returns
+    -------
+    Duration: str
+        Total estimate time in HH:MM:SS format.
+    """
+    # Loading time for whisper + diarization pipeline
+    PIPELINE_LOADING_TIME="00:05:00" 
+    # Loading a 60 minute audio file takes ~5 seconds. This is an upper limit (equivalent to
+    # loading a 24h file) to ensure sufficient time.
+    AUDIO_LOADING_TIME="00:01:00"   
+
+    total_duration = "00:00:00"
+    total_loading = "00:00:00"
+
+    if Path(input_path).suffix == ".json":
+        with open(input_path, "r") as fin:
+            input_files = json.load(fin)
+    else:
+        input_files=input_path
+        
+    for audio_file in input_files:
```

Comment:
This bugs me a bit: if `input_path` does not end in ".json" then is it a string? In that case, would `for audio_file in input_files` loop over the characters in that string?

Reply:
Thanks for the heads up.
`input_path` is not a string if the input is file, but it's `PosixPath`. However `PosixPath` is not iterable and it raises an error.
Fix the bug  [532177c](https://github.com/AaltoRSE/speech2text/pull/3/commits/532177c0a8cd843495fb574cf70307b3745dfc8e).